src/canopy/llm/openai.py

```diff
@@ -91,15 +93,19 @@ def chat_completion(self,
         """  # noqa: E501
 
         model_params_dict: Dict[str, Any] = deepcopy(self.default_model_params)
-        model_params_dict.update(
-            model_params or {}
-        )
+        if self.request_params_passthrough:
+            passthrough_params = {
+                k: v for k, v in (model_params or {}).items() if v is not None
+            }
+            model_params_dict.update(passthrough_params)
+        if model_params_dict.get("model", None) is None:
+            model_params_dict["model"] = self.model_name
```

Comment:
You can probably do that already on `__init()__`:
```python
        super().__init__(model_name)
        ...
        self.default_model_params = kwargs
        self.default_model_params["model] = self.model_name
```
And then the `update()` above would suffice

Reply:
Good suggestion!